docs/source/tutorial/word2vec.rst

```diff
@@ -0,0 +1,346 @@
+
+3.1 Explanation using specific example
+---------------------------------------
+
+In this example, we use the following setups.
+
+* The number of vocabulary :math:`N` is 10.
+* The size of distributed representation vector :math:`D` is 2.
+* Center word is **dog**.
+* Context word is **animal**.
+
+Since there should be more than one Context Word, repeat the following process for each Context Word.
+
+1. The one-hot vector of **dog** is ``[0 0 1 0 0 0 0 0 0 0]`` and you input it as
+   Center Word.
+2. After that, the third row of distributed representation matrix :math:`W_H`
+   for Center Word is the distributed representation of **dog** :math:`L_H`.
+3. The output layer :math:`L_O` is the result of multiplying the distributed 
+   representation matrix :math:`W_O` for Context Words by the distributed
+   representation of **dog** :math:`L_H`.
+4. In order to limit the value of each element of the output layer, 
+   softmax function is applied to the output layer :math:`L_O` to calculate
+   :math:`softmax(L_O)`.
+
+        * To update the parameters, it is necessary to back-propagete the error
+          between the output layer and Context Word.
+        * However, the value of each element of the output layer takes the range
+          :math:`[-\infty, +\infty]`. Context Word's one-hot vector takes only
+          the range :math:`[0, 1]` for each element like ``[1 0 0 0 0 0 0 0 0 0]``.
+        * To limit the value of each element of the output layer between
+          :math:`[0, 1]`, the softmax functions is applied to the layer
+          because the function limits the value between :math:`[0, 1]`.
+
+5. Calculate the error between :math:`W_O` and **animal**'s one-hot vector
+   ``[1 0 0 0 0 0 0 0 0 0 0]``, and propagate the error back to the network
+   to update the parameters.
+
+.. image:: ../../image/word2vec/skipgram_detail.png
+
+4. Implementation of Skip-gram by Chainer
+==========================================
+
+* There is an example related to Word2vec on the GitHub repository, so we will explain based on that.
+
+        * `chainer/examples/word2vec <https://github.com/chainer/chainer/tree/master/examples/word2vec>`_
+
+4.1 Implementation method
+--------------------------
+
+Import package
+^^^^^^^^^^^^^^^
+
+.. literalinclude:: ../../../examples/word2vec/train_word2vec.py
+   :language: python
+   :lines: 12-20
+   :caption: train_word2vec.py
+   :lineno-match:
+
+* Basically if you use chainer you import in this way.
+* Importing functions like F, links like L makes it easy to use.
+
+Define network structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. literalinclude:: ../../../examples/word2vec/train_word2vec.py
+   :language: python
+   :lines: 82-100
+   :caption: train_word2vec.py
+   :lineno-match:
+
+* Next, we define the network structures of skip-gram.
+* When we call the constructor ``__init__``, we pass the vocabulary size
+  ``n_vocab``, the size of the distributed vector ``n_units`` and the loss function
```

Comment:
distributed vector ``n_units``,  and the loss function

Long live the Oxford comma!